nssp_demo/collate_score_tables.R

```diff
@@ -0,0 +1,212 @@
+  table_path <- fs::path(model_run_dir,
+    "score_table",
+    ext = "rds"
+  )
+  location <- fs::path_file(model_run_dir)
+  if (!(fs::file_exists(table_path))) {
+    warning(glue::glue(
+      "No `score_table.rds` found for location ",
+      "{location} in directory {model_run_dir}"
+    ))
+    return(NULL)
+  }
+  score_table <- readRDS(table_path)
+  score_table$quantile_scores$location <- location
+  score_table$sample_scores$location <- location
+  return(score_table)
+}
+
+process_date_score_table <- function(date_fit_dir) {
+  table_path <- fs::path(date_fit_dir,
+    "score_table",
+    ext = "rds"
+  )
+  table_dir <- fs::path_file(date_fit_dir)
+
+  report_date <- stringr::str_match(
+    table_dir,
+    "r_(([0-9]|-)+)_f"
+  )[2] |>
+    lubridate::ymd()
+
+  score_table <- readRDS(table_path)
+  score_table$quantile_scores$report_date <- report_date
+  score_table$sample_scores$report_date <- report_date
+  return(score_table)
+}
+
+bind_tables <- function(list_of_table_pairs) {
+  sample_metrics <- purrr::map(
+    list_of_table_pairs,
+    \(x) {
+      attr(x$sample_scores, "metrics")
+    }
+  ) |>
+    unlist() |>
+    unique()
+  quantile_metrics <- purrr::map(
+    list_of_table_pairs,
+    \(x) {
+      attr(x$quantile_scores, "metrics")
+    }
+  ) |>
+    unlist() |>
+    unique()
+
+  sample_scores <- purrr::map(
+    list_of_table_pairs,
+    \(x) {
+      x$sample_scores
+    }
+  ) |>
+    data.table::rbindlist(fill = TRUE)
+
+  quantile_scores <- purrr::map(
+    list_of_table_pairs,
+    \(x) {
+      x$quantile_scores
+    }
+  ) |>
+    data.table::rbindlist(fill = TRUE)
+
+  attr(sample_scores, "metrics") <- sample_metrics
+  attr(quantile_scores, "metrics") <- quantile_metrics
+
+
+  return(list(
+    sample_scores = sample_scores,
+    quantile_scores = quantile_scores
+  ))
+}
+
+collate_scores_for_date <- function(model_run_dir,
+                                    score_file_name = "score_table",
+                                    score_file_ext = "rds",
+                                    save = FALSE) {
+  message(glue::glue("Processing scores from {model_run_dir}..."))
+  locations_to_process <- fs::dir_ls(model_run_dir,
+    type = "directory"
+  )
+  location_score_tables <- purrr::map(
+    locations_to_process,
+    process_loc_date_score_table
+  )
+
+  date_score_table <- bind_tables(location_score_tables)
+
+  if (save) {
+    save_path <- fs::path(model_run_dir,
+      score_file_name,
+      ext = score_file_ext
+    )
+    message(glue::glue("Saving score table to {save_path}..."))
+    saveRDS(date_score_table, save_path)
+  }
+  message(glue::glue("Done processing scores for {model_run_dir}."))
+  return(date_score_table)
+}
+
+
+collate_all_score_tables <- function(model_base_dir,
+                                     disease,
+                                     score_file_name = "score_table",
+                                     score_file_ext = "rds",
```

Comment:
prefer `score_file_path` to allow for saving in directories other than `model_base_dir` and prevent unintentionally supplying an extension to `score_file_name`

Reply:
In 4ba28c1 I replace `score_file_name`, `score_file_ext`, and save with `score_file_save_path`, default `NULL`, and use it not being `NULL` as the switch for whether to save.